tests/swarm/test_swarm_task.py

```diff
@@ -15,7 +24,7 @@ def test_swarmtask_resources_integration(tool, task_template, db_engine):
     task = task_template.create_task(
         arg="echo qux",
         compute_resources={"cores": 10, "queue": "null.q"},
-        resource_scales={"cores": 0.5},
+        resource_scales=scales,
```

Comment:
would love to see if this test can do more than one rescale, either on the same task or on two different tasks with the same resource scales passed. 

Idea is to confirm that your copying mechanism works as expected, and that the iterator preserves the expected stateful behavior

Reply:
Added another rescale!